README.md
# border_bottom_navigation_bar

A fully customisable Flutter Border Bottom Navigation Bar.

<div style="display:flex">
<img width="355" alt="alert2" src="https://user-images.githubusercontent.com/44444254/158521966-c5567fb6-05f9-4e33-945f-49b9693a2d8b.png" width="200">
<div/>

<b>Usage : </b>

                    class HomeView extends StatefulWidget {
                    @override
                    State<HomeView> createState() => _HomeViewState();
                    }

                    class _HomeViewState extends State<HomeView> {
                    int _currentIndex = 0;
                    @override
                    Widget build(BuildContext context) {
                        return Scaffold(
                        extendBody: true,
                        appBar: AppBar(
                            backgroundColor: Colors.blue[900],
                        ),
                        body: ListView.builder(
                            itemCount: 100,
                            itemBuilder: (context, index) {
                            return Container(
                                height: 300,
                                margin: const EdgeInsets.only(bottom: 10),
                                decoration: BoxDecoration(
                                color: Colors.grey[350],
                                image: DecorationImage(
                                    image: NetworkImage(
                                    'https://picsum.photos/id/$index/500',
                                    ),
                                    fit: BoxFit.cover,
                                ),
                                ),
                            );
                            },
                        ),
                        bottomNavigationBar: BorderBottomNavigationBar(
                            height: 53,
                            currentIndex: _currentIndex,
                            borderRadiusValue: 25,
                            onTap: (index) {
                            setState(() {
                                _currentIndex = index;
                            });
                            print(index);
                            },
                            selectedLabelColor: Colors.white,
                            unselectedLabelColor: Colors.grey,
                            selectedBackgroundColor: Colors.blue[900]!,
                            unselectedBackgroundColor: Colors.grey[200]!,
                            unselectedIconColor: Colors.grey[600]!,
                            selectedIconColor: Colors.white,
                            customBottomNavItems: [
                            BorderBottomNavigationItems(
                                icon: Icons.home,
                            ),
                            BorderBottomNavigationItems(
                                icon: Icons.event_note_outlined,
                            ),
                            BorderBottomNavigationItems(
                                icon: Icons.add_circle_outline_outlined,
                            ),
                            BorderBottomNavigationItems(
                                icon: Icons.rss_feed_outlined,
                            ),
                            BorderBottomNavigationItems(
                                icon: Icons.more_horiz,
                            ),
                            ],
                        ),
                        );
                    }
                    }



CHANGELOG.md
- 0.0.4: Initial version
- 0.0.3: Initial version
- 0.0.2: Initial version
- 0.0.1: Initial version


lib/border_bottom_navigation_bar.dart
library border_bottom_navigation_bar;

import 'package:flutter/material.dart';

class BorderBottomNavigationBar extends StatelessWidget {
  final double height;
  final double borderRadiusValue;
  final List<BorderBottomNavigationItems> customBottomNavItems;

  final Color selectedBackgroundColor;
  final Color unselectedBackgroundColor;
  final bool hideLabel;
  final String? label;
  final double selectedLabelFontSize;
  final double unselectedLabelFontSize;
  final Color selectedLabelColor;
  final Color unselectedLabelColor;
  final Color selectedIconColor;
  final Color unselectedIconColor;
  final int currentIndex;
  final Function(int) onTap;

  const BorderBottomNavigationBar({
    Key? key,
    this.height = 55,
    this.borderRadiusValue = 25,
    required this.customBottomNavItems,
    required this.selectedBackgroundColor,
    required this.unselectedBackgroundColor,
    required this.selectedIconColor,
    required this.unselectedIconColor,
    this.hideLabel = true,
    this.currentIndex = 0,
    this.unselectedLabelFontSize = 12,
    this.selectedLabelFontSize = 14,
    this.selectedLabelColor = Colors.black,
    this.unselectedLabelColor = Colors.black,
    required this.onTap,
    this.label,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Padding(
        padding: EdgeInsets.symmetric(
            horizontal: MediaQuery.of(context).size.width * 0.1),
        child: Container(
          height: height,
          margin: const EdgeInsets.only(bottom: 20),
          clipBehavior: Clip.antiAlias,
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(borderRadiusValue),
          ),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: List.generate(customBottomNavItems.length, (index) {
              return Expanded(
                child: BorderBottomNavigationButton(
                  onTap: onTap,
                  unselectedIconColor: unselectedIconColor,
                  selectedIconColor: selectedIconColor,
                  selectedLabelColor: selectedLabelColor,
                  unselectedLabelColor: unselectedLabelColor,
                  unselectedBackgroundColor: unselectedBackgroundColor,
                  selectedBackgroundColor: selectedBackgroundColor,
                  icon: customBottomNavItems[index].icon,
                  // label: customBottomNavItems[index].label,
                  hideLabel: hideLabel,
                  index: index,
                  currentIndex: currentIndex,
                ),
              );
            }),
          ),
        ),
      ),
    );
  }
}

class BorderBottomNavigationItems {
  final IconData icon;
  // final String label;

  BorderBottomNavigationItems({
    required this.icon,
    // required this.label,
  });
}

class BorderBottomNavigationButton extends StatelessWidget {
  final IconData icon;
  final Color selectedBackgroundColor;
  final Color unselectedBackgroundColor;
  final bool hideLabel;
  final String? label;
  final double selectedLabelFontSize;
  final double unselectedLabelFontSize;
  final Color selectedLabelColor;
  final Color unselectedLabelColor;
  final Color selectedIconColor;
  final Color unselectedIconColor;
  final Function(int) onTap;
  final int? currentIndex;
  final int index;

  const BorderBottomNavigationButton({
    Key? key,
    required this.icon,
    required this.onTap,
    this.currentIndex,
    required this.selectedBackgroundColor,
    required this.unselectedBackgroundColor,
    this.hideLabel = true,
    this.unselectedLabelFontSize = 12,
    this.selectedLabelFontSize = 14,
    this.selectedLabelColor = Colors.black,
    this.unselectedLabelColor = Colors.black,
    required this.index,
    this.label,
    required this.selectedIconColor,
    required this.unselectedIconColor,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      color: currentIndex == index
          ? selectedBackgroundColor
          : unselectedBackgroundColor,
      margin: const EdgeInsets.only(right: 0.2),
      child: Stack(
        alignment: Alignment.center,
        children: [
          Align(
            alignment: Alignment.center,
            child: hideLabel
                ? Icon(
                    icon,
                    color: currentIndex == index
                        ? selectedIconColor
                        : unselectedIconColor,
                  )
                : Column(
                    children: [
                      Expanded(
                        child: Icon(
                          icon,
                          color: currentIndex == index
                              ? selectedIconColor
                              : unselectedIconColor,
                        ),
                      ),
                      Padding(
                        padding: const EdgeInsets.only(bottom: 5.0),
                        child: Text(
                          label!,
                          style: TextStyle(
                            color: currentIndex == index
                                ? selectedLabelColor
                                : unselectedLabelColor,
                            fontSize: currentIndex == index
                                ? selectedLabelFontSize
                                : unselectedLabelFontSize,
                          ),
                        ),
                      ),
                    ],
                  ),
          ),
          Positioned.fill(
            child: Material(
              color: Colors.transparent,
              child: InkWell(
                onTap: () {
                  onTap(index);
                },
              ),
            ),
          )
        ],
      ),
    );
  }
}


example/lib/main.dart
import 'package:border_bottom_navigation_bar/border_bottom_navigation_bar.dart';
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: HomeView(),
    );
  }
}

// ignore: use_key_in_widget_constructors
class HomeView extends StatefulWidget {
  @override
  State<HomeView> createState() => _HomeViewState();
}

class _HomeViewState extends State<HomeView> {
  int _currentIndex = 0;
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      extendBody: true,
      appBar: AppBar(
        backgroundColor: Colors.blue[900],
      ),
      body: ListView.builder(
        itemCount: 100,
        itemBuilder: (context, index) {
          return Container(
            height: 300,
            margin: const EdgeInsets.only(bottom: 10),
            decoration: BoxDecoration(
              color: Colors.grey[350],
              image: DecorationImage(
                image: NetworkImage(
                  'https://picsum.photos/id/$index/500',
                ),
                fit: BoxFit.cover,
              ),
            ),
          );
        },
      ),
      bottomNavigationBar: BorderBottomNavigationBar(
        height: 53,
        currentIndex: _currentIndex,
        borderRadiusValue: 25,
        onTap: (index) {
          setState(() {
            _currentIndex = index;
          });
          print(index);
        },
        selectedLabelColor: Colors.white,
        unselectedLabelColor: Colors.grey,
        selectedBackgroundColor: Colors.blue[900]!,
        unselectedBackgroundColor: Colors.grey[200]!,
        unselectedIconColor: Colors.grey[600]!,
        selectedIconColor: Colors.white,
        customBottomNavItems: [
          BorderBottomNavigationItems(
            icon: Icons.home,
          ),
          BorderBottomNavigationItems(
            icon: Icons.event_note_outlined,
          ),
          BorderBottomNavigationItems(
            icon: Icons.add_circle_outline_outlined,
          ),
          BorderBottomNavigationItems(
            icon: Icons.rss_feed_outlined,
          ),
          BorderBottomNavigationItems(
            icon: Icons.more_horiz,
          ),
        ],
      ),
    );
  }
}
